pyro/infer/autoguide/guides.py

```diff
@@ -396,6 +435,10 @@ def __call__(self, *args, **kwargs):
         if self.prototype_trace is None:
             self._setup_prototype(*args, **kwargs)
 
+        # Needed to capture autoguide parameters as pyro.param statements in the
+        # guide trace
+        pyro.module(self.prefix, self)
```

Comment:
Can we move this line below `self.sample_latent`, to avoid out-of-order issues with #2130

Reply:
(obsolete if we instead use `ConstrainedModule`)